tests/unit_tests/svgtools/web.cpp

```diff
@@ -0,0 +1,133 @@
+/** Detray library, part of the ACTS project (R&D line)
+ *
+ * (c) 2023 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+// Project include(s)
+#include "detray/core/detector.hpp"
+#include "detray/detectors/create_toy_geometry.hpp"
+#include "detray/intersection/detail/trajectories.hpp"
+#include "detray/plugins/svgtools/illustrator.hpp"
+#include "detray/plugins/svgtools/utils/groups.hpp"
+#include "detray/plugins/svgtools/writer.hpp"
+#include "tests/common/tools/particle_gun.hpp"
+
+// Vecmem include(s)
+#include <vecmem/memory/host_memory_resource.hpp>
+
+// Actsvg include(s)
+#include "actsvg/core.hpp"
+#include "actsvg/web/web_builder.hpp"
+
+// System include(s)
+#include <array>
+#include <filesystem>
+#include <string>
+
+int main(int, char**) {
+
+    // In this test we will create a web page to show the detector geometry and
+    // more. We will start by creating the svgs we want to include in the web
+    // page.
+
+    // Axes.
+    const auto axes = actsvg::draw::x_y_axes("axes", {-250, 250}, {-250, 250},
+                                             actsvg::style::stroke());
+
+    // Creating the views.
+    const actsvg::views::x_y view;
+
+    // Creating the detector and geomentry context.
+    using detector_t = detray::detector<detray::toy_metadata<>>;
+    vecmem::host_memory_resource host_mr;
+    const auto [det, names] = detray::create_toy_geometry(host_mr, 4, 3);
+    detector_t::geometry_context context{};
+
+    using transform3_t = typename detector_t::transform3;
+    using vector3 = typename detector_t::vector3;
+
+    // Creating the svg generator for the detector.
+    const detray::svgtools::illustrator il{det, context};
+
+    // The vector of svgs that we want to include on the webpage.
+    std::vector<actsvg::svg::object> svgs;
+
+    // Indexes of the volumes in the detector to be visualized.
+    std::array indices{0UL,  1UL,  2UL,  3UL,  4UL,  5UL,  6UL,
+                       7UL,  8UL,  9UL,  10UL, 11UL, 12UL, 13UL,
+                       14UL, 15UL, 16UL, 17UL, 18UL, 19UL};
+
+    // Draw the volumes and include them in the svg vector.
+    for (std::size_t i : indices) {
+        std::string name = "Volume_" + std::to_string(i);
+        const auto svg = il.draw_volume(name, i, view);
+        svgs.push_back(svg);
+    }
+
+    // Draw the grids and include them in the svg vector.
+    for (const auto i : indices) {
+        if (i % 2 == 0) {
```

Comment:
Hm, maybe this can be done more elegantly by returning an ```std::optional``` from the grid conversion in case a volume does not have a grid?